src/Runner/Linter.php

```diff
@@ -122,11 +137,36 @@ public function run(iterable $files, Ruleset $ruleset, ?FixerInterface $fixer =
             }
             restore_error_handler();
 
-            $rules = $ruleset->getRules();
             foreach ($rules as $rule) {
                 $rule->lintFile($stream, $report, $ignoredViolations);
             }
 
+            if ([] !== $nodeVisitorRules) {
+                $this->setErrorHandler($report, $filePath);
+                try {
+                    $twigSource = new Source($content, $filePath);
+                    $node = $this->env->parse($this->env->tokenize($twigSource));
+                } catch (Error $error) {
+                    $violation = new Violation(
+                        Violation::LEVEL_FATAL,
+                        sprintf('File is invalid: %s', $error->getRawMessage()),
+                        $filePath,
+                        null,
+                        new ViolationId(line: $error->getTemplateLine())
+                    );
+
+                    $report->addViolation($violation);
+                    continue;
+                }
+                restore_error_handler();
+
+                foreach ($nodeVisitorRules as $nodeVisitor) {
+                    $nodeVisitor->lintFile($report, $filePath, $ignoredViolations);
```

Comment:
Since the real `lint` process is done during the `traverse` call, I think we should not call this `lintFile`.

Best I have so far would be `init`/`setFile`/`enterFile` but I dunno, maybe you have something better in mind ?